compiler/src/codegen.rs

```diff
@@ -355,7 +355,7 @@ impl<'a> MethodGen<'a> {
     fn write_service(&self, w: &mut CodeWriter) {
         let req_stream_type = format!("{}<{}>", fq_grpc("RequestStream"), self.input());
         let (req, req_type, resp_type) = match self.method_type().0 {
-            MethodType::Unary => ("req", self.input(), "UnarySink"),
+            MethodType::Unary => ("req", format!("&{}", self.input()), "UnarySink"),
```

Comment:
I don't think you need to update this. Server may need to move the request for taking its bytes or child message.